src/components/Search/Search.js

```diff
@@ -0,0 +1,105 @@
+import React, { useMemo, useState } from 'react';
+import Autosuggest from 'react-autosuggest';
+import {
+  FaSearch,
+  FaChevronCircleDown,
+  FaChevronCircleUp,
+} from 'react-icons/fa';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+
+import SearchService from './SearchService';
+import SearchToggle from './SearchToggle';
+
+import './Search.scss';
+
+function Search() {
+  const dispatch = useDispatch();
+  const activeTopic = useSelector(state => state.app.activeTopic);
+  const [suggestions, setSuggestions] = useState([]);
+  const [value, setValue] = useState('');
+  const { t } = useTranslation();
+
+  const searchService = useMemo(() => {
+    const clear = () => setSuggestions([]);
+    const upsert = (section, items) =>
+      setSuggestions(oldSuggestions => {
+        const index = oldSuggestions.findIndex(s => s.section === section);
+        const newSuggestions = [...oldSuggestions];
+        if (index === -1 && items) {
+          newSuggestions.push({ section, items });
+        } else if (items) {
+          newSuggestions[index] = { section, items };
+        }
+        return newSuggestions;
+      });
+    return new SearchService(activeTopic, clear, upsert, dispatch);
+  }, [activeTopic, setSuggestions, dispatch]);
```

Comment:
`setSuggestion` and `dispatch` changes on every render, aren't they?  Can you verify that the `useMemo` does it's work properly.

Reply:
I'll check...

Reply:
Seems to work, at least a new SearchService is only initilized if activeTopic changes.